Develop dual fix dual mode

haiwentelink opened the pull request.

liucoldstar commented on tlsr9/ble/vendor/controller/tlx/tlx_bt_flash.c:
```diff
@@ -108,9 +114,21 @@ _attribute_no_inline_ int tlx_bt_blc_mac_init(uint8_t *bt_mac)
 	err = flash_write(flash_device, FIXED_PARTITION_OFFSET(vendor_partition)
 			+ TLX_BT_MAC_ADDR_OFFSET, temp_mac, BLE_ADDR_LEN + 3);
 #else
+	#if 0
```
#if 0 #endif这块代码可以修改一下。我会给出个更新后的代码。

haiwentelink replied:
suggest just delete .

liucoldstar commented on tlsr9/drivers/B92/i2c.c:
```diff
@@ -202,7 +202,7 @@ static inline unsigned char i2c_master_data_nack_detect(void)
 {
 	i2c_clr_irq_status(I2C_TX_BUF_STATUS);
-	reg_i2c_id = id & (~FLD_I2C_WRITE_READ_BIT); //BIT(0):R:High  W:Low
+	reg_i2c_id = id; //BIT(0):R:High  W:Low
```
I2C的这块的需求，不同的客户有不同的需求。需要确认一下是否一定在双模SDK中把I2C的改成更develop分支代码不同？最好还是更devlope分支的对齐。

haiwentelink replied:
这个要不整合到一个特殊的i2c的宏里面，在matter里面来控制一下

haiwentelink replied:
类似于 CONFIG_I2C_LED

haiwentelink committed:
riscv: telink: add dual mode logic .

- add analog register proc.
- add extra pwm interface as pointer.
- fix pwm start ,or it will inflect pulse.
- use i2c interface , need raw id .
- fix random mac address .

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
